### ISSUE TITLE
Import error with Python 3.11

### ISSUE THREAD

**Issue by tbutler-qontigo** [pylint-dev/pylint#9743]
### Bug description

```python
Using python 3.11 and any version 3 pylint there is an import error:

from pylint.interfaces import IAstroidChecker, ITokenChecker
ImportError: cannot import name 'IAstroidChecker' from 'pylint.interfaces' (/venv/lib/python3.11/site-packages/pylint/interfaces.py)
```
Using pylint < 3 works and I have heard that using e.g. python 3.9 also works
```


### Configuration

_No response_

### Command used

```shell
python -m pylint app  --rcfile=../.pylintrc
```

### Pylint output

```python
from pylint.interfaces import IAstroidChecker, ITokenChecker
ImportError: cannot import name 'IAstroidChecker' from 'pylint.interfaces' (/venv/lib/python3.11/site-packages/pylint/interfaces.py)
```

### Expected behavior

pylint should work

### Pylint version

```shell
pylint 3.2.3
```

**Comment by mbyrnepr2** [comment]
Hey. Try upgrading astroid which is a dependency of Pylint. Do you know which version of astroid is in your environment?

**Comment by tbutler-qontigo** [comment]
I am running on a standard github actions linux runner which apparently is using 2.3.17.0

**Comment by Pierre-Sassoulas** [comment]
Closing as astroid's version should be managed by pylint, setting/hard coding the astroid version is going to create issues. (If you encounter this problem don't set up astroid or set it to a version that is compatible with pylint after letting a dependency manager solve this for you).

**Comment by tbutler-qontigo** [comment]
@Pierre-Sassoulas I am not trying to change the vwersion of asteroid - that was someone else's suggestion.
I am simply running on the standard github action runner with whatever version it comes with - apparently `2.3.17.0`
How can pylint be made to work with python 11 on a standard `ubuntu-latest` github actions runner?

thanks

**Comment by tbutler-qontigo** [comment]
I'm not sure github would make a fix just for this one app. 
How would I set the asteroid version?  It isn't something I am really familiar with - I am more on the devops side trying to get this to work in our CI pipeline than the python programming side.
Thanks

**Comment by Pierre-Sassoulas** [comment]
If you're using pylint ``3.2.3`` then the required astroid should be set **automatically** to 

https://github.com/pylint-dev/pylint/blob/918d2168e15662df90cfb993df100e566f69d418/pyproject.toml#L44

Unless something somewhere change the version of astroid, pylint is set correctly and should bring the proper dependencies on install. If you don't know what tool or script is ruining your day, you can install the proper astroid manually yourself with ``pip install "astroid>=3.2.2,<=3.3.0-dev0"`` (or another python tool than pip). It will work only for pylint ``3.2.3`` and will ruin your day again if the pylint version change.

**Comment by mathiasose** [comment]
For anyone coming across this like I did and still not understanding what the issue is after upgrading, I figured it out eventually:

The stack trace when pylint crashes doesn't highlight it very clearly, but my problem was with one of the plugins enabled in .pylintrc, coming from a different package that was not compatible.

In my .pylintrc there was the line:
```
load-plugins=pylint_django,pytest_only.ext.pylint
```
and I had upgraded `pylint-django` since that was clearly a pylint-related package, but I had missed that the `pytest-only` package also included a pytest plugin that was enabled. So it has to be either upgraded to support the newer pylint, or removed from .pylintrc.
